File: 2024/MCC/QueueAtSchool.cpp
#include <bits/stdc++.h>
using namespace std;
int a,times;
vector<char> str;
string s;
bool flag1;
int index;

int main(){
    cin >> a >> times;
    cin >> s;
    for(int i =0; i<a; i++){
        str.push_back(s[i]);
    }


    while(times--){
        flag1=false;
        index=0;
        while((index < a) && (flag1==false)){
            cout << str[index] << endl;
            if(str[index]=='B'){

                if(str[index+1]!='B'){
                    str[index+1]='B';
                    str[index]='G';
                    index ++;
                }
                else str[index]='B';
                }
            if(index+1 >= a) flag1=true;
            index++; 
            cout << index << endl;    
        }


        for(char x:str){cout << x;}
        cout << endl;
    }



}

File: 2024/MCC/QueueAtSchool.py
# 5 1
# BGGBG
# GBGGB
amount,times = list(map(int,input().split()))
wordl=input()
word = []
for i in wordl: word.append(i)
flag1=False
index = 0
while times > 0:
    flag1=False
    index = 0
    while index < len(word) and flag1==False:
            
            
            if word[index] == "B":
                try:
                    if word[index+1]!="B":
                        word[index+1]="B"
                        word[index]="G"
                        index +=1
                    else:
                        word[index]="B"
                except IndexError:
                    flag1=True
            index += 1
    times -=1


print(''.join(str(x) for x in word))